routes/w3c/group.js

```diff
@@ -40,17 +43,34 @@ module.exports.route = async function route(req, res) {
  * @param {string} groupName
  */
 async function getGroupInfo(groupName) {
-  const cached = cache.get(groupName);
-  if (cached) {
-    return cached;
+  let groupInfo = store.get(groupName);
+  if (groupInfo) {
+    return groupInfo;
   }
 
   const groupId = groups.hasOwnProperty(groupName) && groups[groupName];
   if (!groupId) {
     throw { statusCode: 404, message: `No group with groupName: ${groupName}` };
   }
 
-  const url = new URL(groupId.toString(), "https://api.w3.org/groups/");
+  await getAllGroupInfo();
+
+  groupInfo = store.get(groupName);
+  if (groupInfo) {
+    return groupInfo;
+  }
+  throw { statusCode: 500, message: `Failed to fetch group details.` };
```

Comment:
Is it possible to throw real errors? You could even do just an inner class:
```
class HTTPError  extends Error {
   constructor(statusCode, message){
        this.statusCode = statusCode;
        this.message = message;
   }
}
```

Reply:
Sent https://github.com/marcoscaceres/respec.org/pull/102